discretize/base/base_mesh.py

```diff
@@ -2246,6 +2668,278 @@ def get_edge_inner_product_deriv(
+        invert_model : bool, optional
+            The inverse of *model* is used as the diagnostic property.
+        invert_matrix : bool, optional
+            Returns the inverse of the inner product surface matrix.
+
+        Returns
+        -------
+        function
+            The function handle :math:`\mathbf{F}(\mathbf{u})` which accepts a
+            (``n_faces``) :class:`numpy.ndarray` :math:`\mathbf{u}`. The function
+            returns a (``n_faces``, ``n_params``) :class:`scipy.sparse.csr_matrix`.
+        """
+        try:
+            A = sdiag(self.face_areas)
+        except NotImplementedError:
+            raise NotImplementedError(
+                f"get_face_inner_product_surface_deriv not implemented for {type(self)}"
+            )
+
+        if model is None:
+            tensorType = -1
+        elif is_scalar(model):
+            tensorType = 0
+        elif model.size == self.nF:
+            tensorType = 1
+        else:
+            raise ValueError(
+                "Unexpected shape of tensor: {}".format(model.shape),
+                "Must be scalar or have length equal to total number of faces.",
+            )
+
+        dMdprop = None
+
+        if invert_matrix or invert_model:
+            MI = self.get_face_inner_product_surface(
+                model,
+                invert_model=invert_model,
+                invert_matrix=invert_matrix,
+            )
+
+        if tensorType == 0:  # isotropic, constant
+            ones = sp.csr_matrix(
+                (np.ones(self.nF), (range(self.nF), np.zeros(self.nF))),
+                shape=(self.nF, 1),
+            )
+            if not invert_matrix and not invert_model:
+                dMdprop = A * ones
+            elif invert_matrix and invert_model:
+                dMdprop = sdiag(MI.diagonal() ** 2) * A * ones * sdiag(1.0 / model**2)
+            elif invert_model:
+                dMdprop = A * sdiag(-1.0 / model**2)
+            elif invert_matrix:
+                dMdprop = sdiag(-MI.diagonal() ** 2) * A
+
+        elif tensorType == 1:  # isotropic, variable in space
+            if not invert_matrix and not invert_model:
+                dMdprop = A
+            elif invert_matrix and invert_model:
+                dMdprop = sdiag(MI.diagonal() ** 2) * A * sdiag(1.0 / model**2)
+            elif invert_model:
+                dMdprop = A * sdiag(-1.0 / model**2)
+            elif invert_matrix:
+                dMdprop = sdiag(-MI.diagonal() ** 2) * A
+
+        if dMdprop is not None:
+
+            def innerProductDeriv(v=None):
+                if v is None:
+                    warnings.warn(
+                        "Depreciation Warning: TensorMesh.innerProductDeriv."
+                        " You should be supplying a vector. "
+                        "Use: sdiag(u)*dMdprop",
+                        FutureWarning,
+                        stacklevel=2,
+                    )
```

Comment:
I know these are in the `_fastInnerProduct` functions, but you should just not accept v=None, so change v to required argument, and remove all of these warnings.

Reply:
Sounds good.